packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionHierarchy_def.hpp

```diff
@@ -795,17 +757,71 @@ void createRegionHierarchy(const int maxRegPerProc,
                               regRowImporters,
                               quasiRegRowMaps);
 
-  int noCoarseSmoothing = 1;
-  if (coarseSolverType == "smoother") noCoarseSmoothing = 0;
+  // std::cout << mapComp->getComm()->getRank() << " | Setup smoothers ..." << std::endl;
 
   tmLocal = Teuchos::null;
   tmLocal = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("createRegionHierarchy: SmootherSetup")));
-  for(int levelIdx = 0; levelIdx < numLevels-noCoarseSmoothing; ++levelIdx) {
+  // Only set the smoother up to the last but one level
+  // if we want to use a smoother on the coarse level
+  // we will handle that separately with "coarse solver type"
+  for(int levelIdx = 0; levelIdx < numLevels - 1; ++levelIdx) {
```

Comment:
@lucbv Shouldn't the index for "all levels except the last one" have an upper bound
```suggestion
  for(int levelIdx = 0; levelIdx < numLevels - 2; ++levelIdx) {
```
?

Reply:
That's a fair question, I will double check that and get back to you.
At the moment that would mean doing an extra setup on the coarsest level so probably not too bad but since it's easy we should fix it if it's wrong!

Reply:
You're right. Stupid me!